### PR TITLE
Enable pod support

### PR DESCRIPTION
**Pull request by ZhaoXuqiang**
share a pod with existing container, with `--net container:xxx`
get shared container by compare pid with exist running container's Pid which found from container's state file

Signed-off-by: Zhang Wei <zhangwei555@huawei.com>
Signed-off-by: ZhaoXuqiang <zhaoxuqiang@huawei.com>

### CODE REVIEW

**Comment by WeiZhang555** [comment]
Please modify your commit message to describe more about this modifcations, also remove "cherry-pick from cfd4695 and 81d42e4", 81d42e4 isn't included in these codes and is outdated so you don't need to mention it. @ZhaoXuqiang

File: cli/create.go [hunk 1]
```diff
@@ -95,20 +97,11 @@ func cmdCreateContainer(context *cli.Context, attach bool) error {
...
-				sharedContainer = ns.Path
-				_, err = os.Stat(filepath.Join(root, sharedContainer, stateJSON))
-				if err != nil {
-					return fmt.Errorf("The container %q is not existing or not ready", sharedContainer)
-				}
-				_, err = os.Stat(filepath.Join(root, sharedContainer, "namespace"))
-				if err != nil {
-					return fmt.Errorf("The container %q is not ready", sharedContainer)
+				if sharedContainer, err = findSharedContainer(context.GlobalString("root"), ns.Path); err != nil {
+					return fmt.Errorf("failed to find shared container: %v", err)
```
Comment (WeiZhang555) [inline cli/create.go]: Can you keep 
```
-				_, err = os.Stat(filepath.Join(root, sharedContainer, stateJSON))
 -				if err != nil {
 -					return fmt.Errorf("The container %q is not existing or not ready", sharedContainer)
 -				}
 -				_, err = os.Stat(filepath.Join(root, sharedContainer, "namespace"))
 -				if err != nil {
 -					return fmt.Errorf("The container %q is not ready", sharedContainer)
```

This piece of code is used for validation, I think it's still useful.

Comment (ZhaoXuqiang) [reply cli/create.go]: it's not useful.
```
 	if sharedContainer != "" {
 		scState, err = loadStateFile(root, sharedContainer)
 		if err != nil {
 			return err
 		}
 		vm, lockFile, err = getSandbox(filepath.Join(context.GlobalString("root"), sharedContainer, "sandbox"))
 		if err != nil {
 			return err
 		}
 	}
```
```
func loadStateFile(root, container string) (*State, error) {
	stateFile := filepath.Join(root, container, stateJSON)
	file, err := os.Open(stateFile)
	if err != nil {
		return nil, err
	}
	defer file.Close()

	var state State
	if err = json.NewDecoder(file).Decode(&state); err != nil {
		return nil, fmt.Errorf("Decode state file %s error: %s", stateFile, err.Error())
	}
	return &state, nil
}
```
loadStateFile will confirm the state file exist.
namespace file is not used.

Comment (WeiZhang555) [reply cli/create.go]: Looks like it's OK then.

ping @laijs

**Comment by laijs** [comment]
LGTM

**Pull request merged**